src/ILCompiler.ReadyToRun/src/ObjectWriter/SectionBuilder.cs

```diff
@@ -0,0 +1,921 @@
+    {
+        /// <summary>
+        /// No relocation required
+        /// </summary>
+        public const ushort IMAGE_REL_BASED_ABSOLUTE = 0x00;
+
+        /// <summary>
+        /// The 32-bit address without an image base (RVA)
+        /// </summary>
+        public const ushort IMAGE_REL_BASED_ADDR32NB = 0x02;
+
+        /// <summary>
+        /// 32 bit address base
+        /// </summary>
+        public const ushort IMAGE_REL_BASED_HIGHLOW = 0x03;
+
+        /// <summary>
+        /// Thumb2: based MOVW/MOVT
+        /// </summary>
+        public const ushort IMAGE_REL_BASED_THUMB_MOV32 = 0x07;
+
+        /// <summary>
+        /// 64 bit address base
+        /// </summary>
+        public const ushort IMAGE_REL_BASED_DIR64 = 0x0A;
+
+        /// <summary>
+        /// 32-bit relative address from byte following reloc
+        /// </summary>
+        public const ushort IMAGE_REL_BASED_REL32 = 0x10;
+
+        /// <summary>
+        /// Thumb2: based B, BL
+        /// </summary>
+        public const ushort IMAGE_REL_BASED_THUMB_BRANCH24 = 0x13;
+
+        /// <summary>
+        /// Arm64: B, BL
+        /// </summary>
+        public const ushort IMAGE_REL_BASED_ARM64_BRANCH26 = 0x14;
+
+        /// <summary>
+        /// 32-bit relative address from byte starting reloc
+        /// This is a special NGEN-specific relocation type
+        /// for relative pointer (used to make NGen relocation
+        /// section smaller)
+        /// </summary>
+        public const ushort IMAGE_REL_BASED_RELPTR32 = 0x7C;
+
+        /// <summary>
+        /// 32 bit offset from base of section containing target
+        /// </summary>
+        public const ushort IMAGE_REL_SECREL = 0x80;
+
+        /// <summary>
+        /// ADRP
+        /// </summary>
+        public const ushort IMAGE_REL_BASED_ARM64_PAGEBASE_REL21 = 0x81;
+
+        /// <summary>
+        /// ADD/ADDS (immediate) with zero shift, for page offset
+        /// </summary>
+        public const ushort IMAGE_REL_BASED_ARM64_PAGEOFFSET_12A = 0x82;
+
+        /// <summary>
+        /// LDR (indexed, unsigned immediate), for page offset
+        /// </summary>
+        public const ushort IMAGE_REL_BASED_ARM64_PAGEOFFSET_12L = 0x83;
+    }
+
+    /// <summary>
+    /// Opaque key representing a single block.
+    /// </summary>
+    public struct BlockHandle : IEquatable<BlockHandle>, IEqualityComparer<BlockHandle>
+    {
+        /// <summary>
+        /// Internal block handle.
+        /// </summary>
+        private int _value;
+
+        /// <summary>
+        /// Construct the block handle for a given key
+        /// </summary>
+        /// <param name="value">Key value to store in the block handle</param>
+        public BlockHandle(int value)
+        {
+            _value = value;
+        }
+
+        /// <summary>
+        /// True when the block handle is empty.
+        /// </summary>
+        public bool IsNull { get => (_value == default(int)); }
```

Comment:
Nit: This can be simplified as `public bool IsNull => _value == default(int);`

Reply:
Thanks for the suggestion, I have already fixed this locally, will push with the next commit based on additional feedback.